src/Symfony/Component/Serializer/Encoder/YamlEncoder.php

```diff
@@ -22,7 +22,7 @@
  */
 class YamlEncoder implements EncoderInterface, DecoderInterface
 {
-    const FORMAT = 'yaml';
+    const FORMAT = array('yaml', 'yml');
```

Comment:
This constant may be used where changing the value from a scalar to an array could break existing code.

Reply:
I think you mean somewhere instead of somewhere.  
So, if  i let this CONST as it is and add another variable (named like $formatExt) equal to the array yaml & yml ? Would it be more reliable ?

Reply:
I'd suggest to mark the const as deprecated, and add a new private one (eg `private const FORMATS = array(...)`